x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/common_attributes.gen.ts

```diff
@@ -205,3 +321,78 @@ export const RelatedIntegration = z.object({
 
 export type RelatedIntegrationArray = z.infer<typeof RelatedIntegrationArray>;
 export const RelatedIntegrationArray = z.array(RelatedIntegration);
+
+export type InvestigationFields = z.infer<typeof InvestigationFields>;
+export const InvestigationFields = z.object({
+  field_names: z.array(z.string()).min(1),
+});
+
+/**
+ * Defines the interval on which a rule's actions are executed.
+ */
+export type RuleActionThrottle = z.infer<typeof RuleActionThrottle>;
+export const RuleActionThrottle = z.union([
+  z.enum(['no_actions', 'rule']),
+  z.string().regex(/^[1-9]\d*[smhd]$/),
+]);
+
+/**
+ * The action frequency defines when the action runs (for example, only on rule execution or at specific time intervals).
+ */
+export type RuleActionFrequency = z.infer<typeof RuleActionFrequency>;
+export const RuleActionFrequency = z.object({
+  summary: z.boolean(),
+  notifyWhen: z.enum(['onActionGroupChange', 'onActiveAlert', 'onThrottleInterval']),
+  throttle: RuleActionThrottle.nullable(),
+});
+
+export type RuleAction = z.infer<typeof RuleAction>;
+export const RuleAction = z.object({
+  /**
+   * The action type used for sending notifications.
+   */
+  action_type_id: z.string(),
+  /**
+   * Optionally groups actions by use cases. Use `default` for alert notifications.
+   */
+  group: z.string(),
+  /**
+   * The connector ID.
+   */
+  id: z.string(),
+  /**
+   * Object containing the allowed connector fields, which varies according to the connector type.
+   */
+  params: z.object({}).catchall(z.unknown()),
+  uuid: z.string().optional(),
+  alerts_filter: z.object({}).catchall(z.unknown()).optional(),
+  frequency: RuleActionFrequency.optional(),
+});
+
+export type RuleExceptionList = z.infer<typeof RuleExceptionList>;
+export const RuleExceptionList = z.object({
+  /**
+   * ID of the exception container
+   */
+  id: z.string().min(1),
+  /**
+   * List ID of the exception container
+   */
+  list_id: z.string().min(1),
+  /**
+   * The exception type
+   */
+  type: z.enum([
+    'detection',
+    'rule_default',
+    'endpoint',
+    'endpoint_trusted_apps',
+    'endpoint_events',
+    'endpoint_host_isolation_exceptions',
+    'endpoint_blocklists',
+  ]),
```

Comment:
In `packages/kbn-securitysolution-io-ts-list-types/src/common/exception_list/index.ts` we have a union type an an enum type for this enum, can we extract it to its own type in the new schema as well?

```ts
export const exceptionListTypeOrUndefined = t.union([exceptionListType, t.undefined]);
export type ExceptionListType = t.TypeOf<typeof exceptionListType>;
export type ExceptionListTypeOrUndefined = t.TypeOf<typeof exceptionListTypeOrUndefined>;
export enum ExceptionListTypeEnum {
  DETECTION = 'detection', // shared exception list type
  RULE_DEFAULT = 'rule_default', // rule default, cannot be shared
  ENDPOINT = 'endpoint',
  ENDPOINT_TRUSTED_APPS = 'endpoint',
  ENDPOINT_EVENTS = 'endpoint_events',
  ENDPOINT_HOST_ISOLATION_EXCEPTIONS = 'endpoint_host_isolation_exceptions',
  ENDPOINT_BLOCKLISTS = 'endpoint_blocklists',
}
```